### PR TITLE
Status issues

### LINKED ISSUE #4732: fix crab status --summary
currently crab status --summary output is empty (at least in some cases) [1]
shoudl at least have the gransummary printed at the end of crab status --long:
```
Summary:
 * Memory: 1232MB min, 1464MB max, 1251MB ave
 * Runtime: 2:59:11 min, 12:32:49 max, 5:58:04 ave
 * CPU eff: 67% min, 98% max, 95% ave
 * Waste: 27:12:38 (3% of total)
```
[1]
```
belforte@lxplus099/sal> crab status -d crab_JetHT_Run2016D-07Aug17-v1/ --summary 
Warning: Incompatible CRABClient version development
Server is saying that compatible versions are: [u'3.3.1704', u'3.3.1705', u'3.3.1706', u'3.3.1707', u'3.3.1711', u'3.3.1711.rc4']
CRAB project directory:		/afs/cern.ch/work/b/belforte/CRAB3/TC3/dbg/sal/crab_JetHT_Run2016D-07Aug17-v1
Task name:			180115_161640:srappocc_crab_JetHT_Run2016D-07Aug17-v1
Grid scheduler:			crab3@vocms0195.cern.ch
Status on the CRAB server:	SUBMITTED
Task URL to use for HELP:	https://cmsweb.cern.ch/crabserver/ui/task/180115_161640%3Asrappocc_crab_JetHT_Run2016D-07Aug17-v1
Dashboard monitoring URL:	http://dashb-cms-job.cern.ch/dashboard/templates/task-analysis/#user=srappocc&refresh=0&table=Jobs&p=1&records=25&activemenu=2&status=&site=&tid=180115_161640%3Asrappocc_crab_JetHT_Run2016D-07Aug17-v1
Status on the scheduler:	COMPLETED

Jobs status:                    finished      100.0% (146/146)

Publication status:		done           93.8% (137/146)
				failed          6.2% (  9/146)

Output dataset:			/JetHT/srappocc-JetHT_Run2016D-07Aug17-v1-b84139e74cafd5a0bfdaf4226ec7e7b4/USER
Output dataset DAS URL:		https://cmsweb.cern.ch/das/request?input=%2FJetHT%2Fsrappocc-JetHT_Run2016D-07Aug17-v1-b84139e74cafd5a0bfdaf4226ec7e7b4%2FUSER&instance=prod%2Fphys03

Site Summary Table (including retries):

Site                    Runtime      Waste    Running Successful   Stageout     Failed

Log file is /afs/cern.ch/work/b/belforte/CRAB3/TC3/dbg/sal/crab_JetHT_Run2016D-07Aug17-v1/crab.log
belforte@lxplus099/sal> 
```

### PR DESCRIPTION
**Pull request by lecriste**
Along with #4748 resolves #4732, #4733 and #4739.

### CODE REVIEW

File: src/python/CRABClient/Commands/status.py [hunk 1]
```diff
@@ -171,6 +171,10 @@ def makeStatusReturnDict(self, crabDBInfo, combinedStatus, dagStatus = '',
             this dictionary in their scripts.
         """
 
+        if shortResult is None: shortResult = {}
```
Comment (belforte) [inline src/python/CRABClient/Commands/status.py]: I really can't understand why we should replace
```
a={}
```
with
```
a = None
if a is None: a = {}
```
are you really sure that we need this commit ?

**Comment by belforte** [comment]
too many things in this PR, can we rather push the very minimal things ?

File: src/python/CRABClient/Commands/status.py (hunk 1 above)
Comment (lecriste) [reply src/python/CRABClient/Commands/status.py]: It removes the first warning from pylint in the previous Jenkins results:
https://cmssdt.cern.ch/dmwm-jenkins/view/All/job/DMWM-CRABClient-PR-test/399/artifact/artifacts/PullRequestReport.html

**Comment by lecriste** [comment]
Most of them comes from the last two commits which aim at fixing the Jenkins results.
We can decouple the first two commits from the second two although I intended to keep them separte also when merging.
PS: I would merge this once I add also the CPU efficiency monitoring as discussed in the last meeting.

**Comment by belforte** [comment]
@lecriste I know you wanted to fix the Jenkins complain, but honestly I liked more the dangerous version ! Anyhow we have that kind of complain (redefining X from outer scope e.g.) all over CRAB, I'd rather not bother to get rid of them. The risk of touching a line we did not expect to touch is always high, a small typo can slip in and create subtle issues :-(  We can certainly live w/o the need for people to review and test cosmetic changes (imagine e.g. that you forget one of those `if a is None a={}` lines.
Python3 compatibility is a different story, we need to take care.
 So please try to stay minimal in fugure PR's. As the saying goes "if it ain't broken, don't fix it" !

**Comment by belforte** [comment]
@lecriste where are we on this PR ?

**Comment by lecriste** [comment]
@belforte I am introducing also the monitoring of the CPU efficiency (last item [here](https://hypernews.cern.ch/HyperNews/CMS/get/crabDevelopment/2731.html)). Once I commit it we can merge.

**Comment by belforte** [comment]
I do not understand why Jenkins still reports the Pythn3 compatibility problem.
Otherwise this is OK.
let me see if I can force a jenkins re-run, for fun

**Comment by belforte** [comment]
test this

**Comment by lecriste** [comment]
I have a question: as it is now, [waste](https://github.com/lecriste/CRABClient/blob/4a0b0ff3276e93cf2d2e6e2ee9b631dd53aa4d6f/src/python/CRABClient/Commands/status.py#L420) accounts also for internal Condor restarts, e.g.:
http://dashb-cms-job.cern.ch/dashboard/request.py/condorrestarts?jobId=2086237649

Is it so by design or is it intended to represent the time wasted because of the user code/configuration?

**Comment by belforte** [comment]
about "waste".... let's leave it as it is. It is not really used by anyone atm and if we start to we will have to review and possibly breake it down.

**Comment by belforte** [comment]
I am still puzzled that Jenkins still complain about python3 thing that you committed a fix for.. let's plow on. I'll merge.

**Pull request merged**